src/category_theory/sites/cocontinuous.lean

```diff
@@ -0,0 +1,201 @@
+covering sieves. This seems stronger that the definition found in stacks project,
+but they are actually equivalent via `category_theory.grothendieck_topology.superset_covering`
+(The precise statement not formalized, but follows from it quite trivially).
+
+This concept is also called the "covering lifting property" as defined in [MM92] Chapter VII,
+Section 10, and should not be confused with the general definition of cocontinuous functors
+between categories as functors preserving small colimits.
+
+## Main Results
+- `category_theory.sites.cocontinuous.Ran_is_sheaf`: If `F : C ⥤ D` is cocontinuous, then
+`Ran F.op` as a functor `(Cᵒᵖ ⥤ A) ⥤ (Dᵒᵖ ⥤ A)` of presheaves maps sheaves to sheaves.
+
+## References
+
+* [S. MacLane, I. Moerdijk, *Sheaves in Geometry and Logic*][MM92]
+* https://stacks.math.columbia.edu/tag/00XI
+
+-/
+
+universes u
+noncomputable theory
+
+open category_theory
+open opposite
+open category_theory.presieve.family_of_elements
+open category_theory.presieve
+open category_theory.limits
+
+namespace category_theory
+namespace sites
+variables {C D A : Type u} [category.{u} C] [category.{u} D] [category.{u} A] [has_limits A]
+variables {J : grothendieck_topology C} {K : grothendieck_topology D}
+
+/--
+A functor `F : (C, J) ⥤ (D, K)` between sites is called cocontinuous if for all covering sieves
+`R` in `D`, `R.pullback F` is a covering sieve in `C`.
+-/
+structure cocontinuous (J : grothendieck_topology C) (K : grothendieck_topology D) (F : C ⥤ D) :=
+(cover_lift : ∀ {U : C} {S : sieve (F.obj U)} (hS : S ∈ K (F.obj U)), S.functor_pullback F ∈ J U)
+
+namespace cocontinuous
+
+/-- A trivial example to make the inhabited linter happy. -/
+instance id_cocontinuous : inhabited (cocontinuous J J (𝟭 _)) :=
+⟨⟨λ _ S h, by { convert h, simp }⟩⟩
+
+
+variables {F : C ⥤ D} (HF : cocontinuous J K F)
+namespace Ran_is_sheaf
+/-
+We will now prove that `Ran F.op` maps sheaves to sheaves if `F` is cocontinuous. This can be found
+in https://stacks.math.columbia.edu/tag/00XK. However, the proof given here uses the
+amalgamation definition of sheaves, and thus does not require that `C` or `D` has catigorical
+pullbacks.
+
+Suppose there is a compatible family `x` of elements on `U`, we ought to glue them together
+to form a unique section on `S`. We will achieve this by restricting `x` onto objects of the form
+`F(Y)` with `Y : C`, and glueing them via the sheaf property of `ℱ` on `C`.
+The uniqueness also follows from the uniqueness provided by the sheaf property.
+-/
+variable (ℱ : Sheaf J A)
+variables {X : A} {U : D} {S : sieve U} (hS : S ∈ K U)
+variables {x : S.arrows.family_of_elements ((Ran F.op).obj ℱ.val ⋙ coyoneda.obj (op X))}
+variable (hx : x.compatible)
+
+include HF hS hx
+
+/-- Given a `F(Y) ⟶ U`, we can find a unique section in `ℱ(Y)` that agrees with `x` on `Y`. -/
+lemma get_section (Y : structured_arrow (op U) F.op) :
+ ∃! (t : (ℱ.val ⋙ coyoneda.obj (op X)).obj (op (unop Y.right))),
```

Comment:
I think the use of `coyoneda` here (and in other places) makes this difficult to read. Since you're working with amalgamations anyway, have you considered first proving the entire thing for `SheafOfTypes`s and then using `is_sheaf_iff_is_sheaf_of_type` for the general case? I don't have an overview of all the details of your proof, but maybe this could be the cleaner approach?

Reply:
I could try, but I cannot guarantee that it will work. Hopefully it does and we could get rid of the ugly stuff.

Reply:
I've just tried a bit, and The construction is basically the same but merely with `⋙ coyoneda.obj (op X)` removed.
I'm not sure if this is a cleaner approach per se.

Reply:
Conceptually, I find the two-step process easier to understand where you first prove it for type-valued sheaves. But it might not make a big difference.